### PR TITLE
View user profile page

### LINKED ISSUE #6: As an admin I would like to see the user profile of an user I am administrating
User profile
- [ ] Photo (optional, not editable)
- [ ] Name (mandatory, not editable)
- [ ] Email (mandatory, not editable)
- [ ] Phone (optional, not editable)
- [ ] Facebook link (optional, not editable)
- [ ] LinkedIn link (optional, not editable)
- [ ] Notes by Admin or Coordinator (optional, editable)

### PR DESCRIPTION
**Pull request by breta-ionut**
## Issue reference

Resolves #6.

## Small description of change

Implemented the view profile page.

## Type of Change

- [ ] Fix
- [x] Feature
- [ ] Documentation Updates
- [ ] Breaking Changes

## Additional Information

Not needed.

### CODE REVIEW

File: src/components/UserProfile/UserProfileHeaderContent.tsx [hunk 1]
```diff
@@ -11,45 +11,44 @@ import {
...
-  },
-}));
+  }));
 
 interface UserProfileDisplayProps {
   email: string;
   displayName: string;
   profileImageUrl: string;
-  socialInfo: string;
+  socialInfo: SocialInfo | string;
```
Comment (endriu3314) [inline src/components/UserProfile/UserProfileHeaderContent.tsx:38]: What is `| string` used for, shouldn't every request be SocialInfo?

File: src/components/UserProfile/UserProfileHeaderContent.tsx [hunk 2]
```diff
@@ -11,45 +11,44 @@ import {
...
   phone: string;
   roles: number[];
-  switchToEditModeHandler: any;
+  switchToEditModeHandler?: any;
+  children?: JSX.Element | JSX.Element[];
```
Comment (endriu3314) [inline src/components/UserProfile/UserProfileHeaderContent.tsx:42]:
```suggestion
  children: React.ReactNode;
```

File: src/components/UserProfile/UserProfileHeaderContent.tsx [hunk 3]
```diff
@@ -11,45 +11,44 @@ import {
...
 interface UserProfileDisplayProps {
   email: string;
   displayName: string;
   profileImageUrl: string;
-  socialInfo: string;
+  socialInfo: SocialInfo | string;
   phone: string;
   roles: number[];
-  switchToEditModeHandler: any;
+  switchToEditModeHandler?: any;
```
Comment (endriu3314) [inline src/components/UserProfile/UserProfileHeaderContent.tsx:41]:
```suggestion
  switchToEditModeHandler?: React.MouseEventHandler<HTMLButtonElement>;
```

File: src/components/UserProfile/UserProfileHeaderContent.tsx [hunk 4]
```diff
@@ -11,45 +11,44 @@ import {
...
-  const socialInfo: SocialInfo = props.socialInfo
-    ? JSON.parse(props.socialInfo)
-    : { facebook: "", linkedIn: "" };
+const UserProfileHeaderContent = (props: UserProfileDisplayProps) => {
+  const socialInfo: SocialInfo = !props.socialInfo
+      ? { facebook: "", linkedIn: "" }
+      : typeof props.socialInfo === "string"
+      ? JSON.parse(props.socialInfo as string)
+      : props.socialInfo,
+    classes = useStyles();
```
Comment (endriu3314) [inline src/components/UserProfile/UserProfileHeaderContent.tsx:51]: What if we destructure the props and give it the default value `{ facebook: "", linkedIn: ""}`, like this:

```js
const UserProfileHeaderContent: React.FC<UserProfileDisplayProps> = ({
  socialInfo = { facebook: "", linkedIn: "" },
  ...props
})
```

I don't get where should that string come from, I may be wrong, but all this `? : ? :` is really hard to read

File: src/pages/UserProfilePage/UserProfilePage.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+const UserProfilePage: React.FC = () => {
+  const tokenStatus = useTokenStatus(),
+    id: number = useParams().id as unknown as number,
+    [user, setUser] = useState<SpecifiedUser>(),
+    navigate = useNavigate();
+
+  useEffect(() => {
+    const loadUser = async () => {
```
Comment (endriu3314) [inline src/pages/UserProfilePage/UserProfilePage.tsx:26]: Why not use Suspense for this? That's why we upgraded to React 18

**Review by endriu3314 — CHANGES_REQUESTED** [review]

File: src/components/UserProfile/UserProfileHeaderContent.tsx (hunk 1 above)
Comment (breta-ionut) [reply src/components/UserProfile/UserProfileHeaderContent.tsx:39]: This component is used in other 2 components. One of them fetches the user from a service with the social info already deserialized, the other gets it encoded. In order to support both usages, I made this change. I know that ideally we should receive the data in a consistent manner from the API and not as we do now, with JSON strings embedded in JSONs, but fixing these issues right now would get us out of the scope of this feature.

File: src/components/UserProfile/UserProfileHeaderContent.tsx (hunk 4 above)
Comment (breta-ionut) [reply src/components/UserProfile/UserProfileHeaderContent.tsx:52]: I think that your proposal would make the code more readable, however it doesn't properly handle the case when the socialInfo field comes as null from the back-end :(

Comment (breta-ionut) [reply src/components/UserProfile/UserProfileHeaderContent.tsx:52]: Btw, the ES linter doesn't do a great job at formatting this code block.

File: src/pages/UserProfilePage/UserProfilePage.tsx (hunk 5 above)
Comment (breta-ionut) [reply src/pages/UserProfilePage/UserProfilePage.tsx:27]: Unfortunately, from what I understand, simply replacing the current logic by using Suspense isn't that straight-forward. As far as I can tell, Suspense isn't capable of monitoring the pending promises in your component and as they are pending display the fallback. Instead it gets to the fallback when errors are triggered and in order to properly use it with data fetching components you must wrap the data fetching processes in order to throw errors if the data isn't ready. Or at least this is what I understood after reading https://blog.logrocket.com/react-suspense-data-fetching/.

File: src/components/UserProfile/UserProfileHeaderContent.tsx (hunk 4 above)
Comment (endriu3314) [reply src/components/UserProfile/UserProfileHeaderContent.tsx:52]:
```suggestion
  const parseSocialInfo =
      typeof props.socialInfo === "string"
        ? JSON.parse(props.socialInfo as string)
        : props.socialInfo,
    socialInfo: SocialInfo = !props.socialInfo
      ? { facebook: "", linkedIn: "" }
      : parseSocialInfo,
    classes = useStyles();
```

What if we do it like this? Other than this it looks good to me

**Review by endriu3314 — DISMISSED** [review]

File: src/components/UserProfile/UserProfileHeaderContent.tsx (hunk 4 above)
Comment (breta-ionut) [reply src/components/UserProfile/UserProfileHeaderContent.tsx:52]: Looks better indeed, I will apply your suggestion, thanks!

**Review by endriu3314 — APPROVED** [review]
LGTM

**Pull request merged**